velox/functions/lib/tests/QuantileDigestTest.cpp

```diff
@@ -952,4 +976,290 @@ TEST_F(QuantileDigestTest, getDistributionFunction) {
   testGetDistributionFunction<float>();
 }
 
+TEST_F(QuantileDigestTest, getValuesInQuantileRange) {
+  QuantileDigest<int64_t> digest{StlAllocator<int64_t>(allocator()), 0.01};
+
+  digest.add(1, 2.0);
+  digest.add(5, 3.0);
+  digest.add(10, 1.0);
+  digest.add(15, 4.0);
+
+  auto result1 = digest.getValuesInQuantileRange(0.1, 0.7);
+  EXPECT_GT(result1.size(), 0);
+
+  double totalWeight = 0.0;
+  for (const auto& pair : result1) {
+    totalWeight += pair.count;
+  }
+  EXPECT_NEAR(totalWeight, 6.0, 0.1);
+
+  auto result2 = digest.getValuesInQuantileRange(0.2, 0.6);
+  totalWeight = 0.0;
+  double totalValue = 0.0;
+  for (const auto& pair : result2) {
+    totalWeight += pair.count;
+    totalValue += pair.value * pair.count;
+  }
+  EXPECT_NEAR(totalWeight, 4.0, 0.1);
+  EXPECT_NEAR(totalValue, 25.0, 0.1);
+
+  auto result3 = digest.getValuesInQuantileRange(0.5, 0.8);
+  totalWeight = 0.0;
+  for (const auto& pair : result3) {
+    totalWeight += pair.count;
+  }
+  EXPECT_NEAR(totalWeight, 3.0, 0.1);
+
+  auto result4 = digest.getValuesInQuantileRange(0.3, 0.3);
+  EXPECT_EQ(result4.size(), 0);
+
+  QuantileDigest<int64_t> singleDigest{
+      StlAllocator<int64_t>(allocator()), 0.01};
+  singleDigest.add(42, 5.0);
+
+  auto result5 = singleDigest.getValuesInQuantileRange(0.0, 1.0);
+  EXPECT_EQ(result5.size(), 1);
+  EXPECT_EQ(result5[0].value, 42);
+  EXPECT_NEAR(result5[0].count, 5.0, 1e-6);
+}
+
+TEST_F(QuantileDigestTest, getValuesInQuantileRangeWeighted) {
```

Comment:
nit: Is this test case the same as TEST_F(QuantileDigestTest, getValuesInQuantileRange)? If so, let's remove it.

Reply:
Ah, missed while refactoring